compiler+runtime/src/cpp/jank/runtime/core/math.cpp

```diff
@@ -1702,4 +1702,40 @@ namespace jank::runtime
       },
       o);
   }
+
+  native_integer parse_long(object_ptr const o)
+  {
+    return visit_object(
+      [](auto const typed_o) -> native_integer {
+        using T = typename decltype(typed_o)::value_type;
+
+        if constexpr(std::same_as<T, obj::persistent_string>)
```

Comment:
You're visiting *every single object type*. This lambda template will be instantiated 52 times (the number of objects we have). Only one of those does anything useful. In other words visit is not what you want here.

Either check the `o->type` or use `dyn_cast`. Note that C++ doesn't allow direct initialization in this one particular syntax.

```cpp
if(auto const typed_o = dyn_cast<obj::persistent_string>(o))
{
  return std::stoll(typed_o->data);
}
else
{
  throw erase(make_box("Expected string, got <TODO: type>"));
}
```